### ISSUE TITLE
Menu Titles

### ISSUE THREAD

**Issue by JoomFX** [gantry/gantry5#670]
@w00fz  I'm trying to figure it out for more than an hour now but I cannot and I really do not understand why. Please read below and you will understand what I mean.....

As you know there was a discussion in the chat about the Menu Item Titles that appear when you hover over an item. The user asked that those are made optional.
I checked more than 10 of the most popular websites and none of them have Menu Item Titles. So I decided to submit a PR with this feature.

And here comes my actual question. I cannot make it work and I do not understand why. I thought I will do it for less than 5mins but.... it seems that `menu.html.twig` does not understand the variables/values from `menu.yaml`.

Here's the YAML code I use:

```
    enableTitles:
      type: select.selectize
      label: Enable Titles
      description: Enable the titles/tooltips of the Menu Items.
      default: 0
      options:
        0: No
        1: Yes
```

And here's the modification in the Twig:

```
{% if particle.enableTitles|default('0') == '1' %}title="{{ item.title|e }}"{% endif %}
```

But it doesn't work.
Please Djamil, can you tell me what I'm doing wrong? It's driving me crazy mate :)

P.S.   As you can see from my code, I think that the Menu Item Titles should be disabled by default and then if a user needs them, he can enable them from the Menu Particle.

**Comment by w00fz** [comment]
Have you tried like this?

```twig
{% if particle.enableTitles|default(false) %}title="{{ item.title|e }}"{% endif %}
```

**Comment by JoomFX** [comment]
@w00fz I just tried the code you suggested above but it still doesn't work.

Whatever condition/code I use, the Title is gone, but if I set the option to show the Title, it still does not show it. It really seems that the variable is not being recognized. I also did some tests, I tried to just add the variable value in the Titel with the following code but it still does not work:

```
title="{{ item.title|e }} {{ particle.enableTitles|e }}"
```
and
```
title="{{ item.title|e }} {{ particle.enableTitles }}"
```

It really seems that the YAML variable is to being recognized in the Twig. I tried with different variable name, different Twig syntax but nothing seem to work. I'm really lost here - I cannot figure out what it doesn't work. Could you please do some tests?

**Comment by mahagr** [comment]
@lilmike Those titles were added for accessibility so I'd like to hear your feedback, too. :)

**Comment by JoomFX** [comment]
@w00fz @mahagr guys, did you have the change to test this?

I really cannot figure out why the option I'm trying to add does not work. The YAML variable does not seem to be recognized in the Twig.

@mahagr I'm just trying to add an option in the Menu Particle which will make the Menu Titles optional (ON/OFF switch). And I believe they should be OFF by default... but it doesn't really matter :)

**Comment by w00fz** [comment]
I'm not sure how was the entire code of yours @JoomFX but I believe you were in a twig `macro` and tried to use the external variable `particle` which was not available.

It is kind of tricky with twig to do that, can see some docs here: http://twig.sensiolabs.org/doc/tags/macro.html

Anyway this is now implemented and it's set off by default.

**Comment by JoomFX** [comment]
Thanks a lot @w00fz , as usual :)

Yep, that's exactly what I was trying to do.
I just checked [your commit](https://github.com/gantry/gantry5/commit/8f98f9cbe1df15689f4589a4d9ef8869f1801aea) and I saw what you did. Just one question - you've added `particle` to all marcos. Did you do that so from now on all new variables defined in the YAML file will be available in the Twig? I mean, adding `particle` to all these places was not needed in order to implement the `renderTitle` option, right?

I hope you understand what I'm asking :D
